compiler/rustc_query_system/src/query/plumbing.rs

```diff
@@ -254,8 +254,9 @@ where
     Q: QueryConfig<Qcx>,
     Qcx: QueryContext,
 {
-    let error =
-        try_execute.find_cycle_in_stack(qcx.collect_active_jobs(), &qcx.current_query_job(), span);
+    let (query_map, complete) = qcx.collect_active_jobs();
+    assert!(complete, "failed to collect active queries");
```

Comment:
Why is incomplete map insufficient here?
What can happen if we try to `find_cycle_in_stack` with an incomplete map?

Reply:
It will just panic if the relevant queries are missing. It's useful to differentiate the errors though.

Reply:
Could you add this as a comment to code?